contracts/VotingMachines/Governance.sol

```diff
@@ -0,0 +1,509 @@
+        uint noneBoostedVotePeriodLimit; //the time limit for a proposal to be in an absolute voting mode.
+        uint boostedVotePeriodLimit; //the time limit for a proposal to be in an relative voting mode.
+        uint scoreThreshold;
+        GovernanceFormulasInterface governanceFormulasInterface;
+        uint minimumStaketingFee;
+    }
+    struct Voter {
+        uint vote; // 0 - 'abstain'
+        uint reputation; // amount of voter's reputation
+    }
+
+    struct Staker {
+        uint vote; // 0 - 'abstain'
+        uint amount; // amount of voter's reputation
+    }
+
+    struct Proposal {
+        address avatar; // the organization's avatar the proposal is target to.
+        uint numOfChoices;
+        ExecutableInterface executable; // will be executed if the proposal will pass
+        uint totalVotes;
+        uint totalStakes;
+        uint submittedTime;
+        uint boostedPhaseTime; //the time the proposal shift to relative mode.
+        ProposalState state;
+        uint winningVote; //the winning vote.
+        mapping(uint=>uint) votes;
+        mapping(address=>Voter) voters;
+        mapping(uint=>uint) stakes;
+        mapping(address=>Staker) stakers;
+    }
+
+    event NewProposal(bytes32 indexed _proposalId, uint _numOfChoices, address _proposer, bytes32 _paramsHash);
+    event CancelProposal(bytes32 indexed _proposalId);
+    event ExecuteProposal(bytes32 indexed _proposalId, uint _decision);
+    event VoteProposal(bytes32 indexed _proposalId, address indexed _voter, uint _vote, uint _reputation);
+    event CancelVoting(bytes32 indexed _proposalId, address indexed _voter);
+    event Stake(bytes32 indexed _proposalId, address indexed _voter,uint _vote,uint _amount);
+    event Redeem(bytes32 indexed _proposalId, address indexed _beneficiary,uint _amount);
+
+    mapping(bytes32=>Parameters) public parameters;  // A mapping from hashes to parameters
+    mapping(bytes32=>Proposal) public proposals; // Mapping from the ID of the proposal to the proposal itself.
+
+    uint constant MAX_NUM_OF_CHOICES = 10;
+    uint proposalsCnt; // Total amount of proposals
+    mapping(address=>uint) public orgBoostedProposalsCnt;
+    StandardToken public stakingToken;
+
+    /**
+     * @dev Constructor
+     */
+    function Governance(StandardToken _stakingToken)
+    public
+    {
+        stakingToken = _stakingToken;
+    }
+
+  /**
+   * @dev Check that the proposal is votable (open and not executed yet)
+   */
+    modifier votable(bytes32 _proposalId) {
+        require((proposals[_proposalId].state == ProposalState.NoneBoosted) || (proposals[_proposalId].state == ProposalState.Boosted));
+        _;
+    }
+
+    /**
+     * @dev hash the parameters, save them if necessary, and return the hash value
+    */
+    function setParameters(
+        Reputation _reputationSystem, // the reputation system that is being used
+        uint _noneBoostedVoteRequierePercentage, // the absolute vote percentages bar.
```

Comment:
Misspelled, should be _nonBoostedVoteRequiredPercentage?

Reply:
Update at a6ba131